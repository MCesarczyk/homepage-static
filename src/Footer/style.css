.footer__wrapper {
  max-width: 32rem;
  margin: 0 auto;
  padding: 4rem 0;
}

.footer__header {
  font-size: 2rem;
  margin: 0;
}

.footer__caption {
  font-size: 12px;
  font-weight: 700;
  color: #fff;
}

.footer__link {
  display: block;
  font-weight: 900;
  color: #fff;
  margin: 1.5rem 0 1.5rem 0;
  transition: 1s;
}

.footer__link:hover {
  color: #2188ff;
  transform: scale(1.01);
}

.footer__link:active {
  color: unset;
  transform: unset;
}

.footer__paragraph {
  margin-bottom: 3rem;
  text-align: left;
}