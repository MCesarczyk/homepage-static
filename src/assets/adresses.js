export const MAIL_ADRESS = "michal.cesarczyk@gmail.com";
export const GITHUB_ADRESS = "https://github.com/MCesarczyk";
export const LINKEDIN_ADRESS = "https://www.linkedin.com/in/micha%C5%82-cesarczyk-0820589a";
export const UPWORK_ADRESS = "https://www.upwork.com/freelancers/~01ec95abd3619f24c6";
export const CODEPEN_ADRESS = "https://codepen.io/bikeoholic";
export const FACEBOOK_ADRESS = "https://www.facebook.com/mich.cesarz/";
export const INSTAGRAM_ADRESS = "https://www.instagram.com/mich_cesarz/";