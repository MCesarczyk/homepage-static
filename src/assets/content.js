export const sections = ["skills", "portfolio", "goals"];