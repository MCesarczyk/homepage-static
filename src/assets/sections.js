export const sections = [
  {
    id: 1,
    name: "skills",
    title: "My skillset",
    previewType: "list",
    items: [
      {
        id: 1,
        name: "Semantic & responsive web design",
        details: [
          "HTML",
          "CSS3",
          "BEM convention",
          "CSS-Flexbox",
          "CSS-Grid"
        ]
      },
      {
        id: 2,
        name: "Javascript",
        details: [
          "JS/ES6+ features",
          "Objects/Arrays operations",
          "Immutability"
        ]
      },
      {
        id: 3,
        name: "Asynchronic operations",
        details: [
          "Promises",
          "Async/Await",
          "AJAX/API/LocalStorage",
          "Error handling"
        ]
      },
      {
        id: 4,
        name: "React",
        details: [
          "React hooks/custom Hooks",
          "React-Redux/Redux-toolkit",
          "Redux-Saga",
          "React Router"
        ]
      },
      {
        id: 5,
        name: "React app styling",
        details: [
          "Styled components",
          "Ant Design UI library",
          "Material UI library",
          "Bootstrap"
        ]
      },
      {
        id: 6,
        name: "Building fulstack apps",
        details: [
          "Ruby on Rails framework",
          "PostgreSQL basics",
          "Restful API",
          "packet managers: NVM, RVM, Gem bundler"
        ]
      },
      {
        id: 7,
        name: "Deploying apps",
        details: [
          "Git/gh-pages",
          "Digitalocean",
          "Nginx/Passenger",
          "AWS S3 storage handling"
        ]
      },
      {
        id: 8,
        name: "Teamworking",
        details: [
          "GitHub flow, collaborating",
          "Trello: tasks management",
          "Figma: reading UI layout",
          "Scrum/Agile metodology"
        ]
      },
      {
        id: 9,
        name: "Other skills",
        details: [
          "Expertise in CNC programming",
          "CNC machines setting/operation",
          "General technical knowledge"
        ]
      }
    ]
  },
  {
    id: 2,
    name: "portfolio",
    title: "My portfolio",
    previewType: "gallery",
    items: []
  },
  {
    id: 3,
    name: "goals",
    title: "My closest goals",
    previewType: "list",
    items: [
      {
        id: 1,
        name: "Things I'm currently learning",
        details: [
          "Testing code: Jest/Enzyme",
          "TypeScript"
        ]
      },
      {
        id: 2,
        name: "Things I plan to learn subsequently",
        details: [
          "Node.js",
          "GraphQL",
          "Next.js"
        ]
      }
    ]
  }
];