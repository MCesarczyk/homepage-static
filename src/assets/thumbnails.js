import { ReactComponent as GithubThumbnail } from "./githubThumbnail.svg";
import { ReactComponent as LinkedinThumbnail } from "./linkedinThumbnail.svg";
import { ReactComponent as UpworkThumbnail } from "./upworkThumbnail.svg";

import {
    GITHUB_ADRESS,
    LINKEDIN_ADRESS,
    UPWORK_ADRESS,
} from "./adresses";

export const thumbnails = [
    {
        id: 1,
        testId: "github-icon-test",
        icon: <GithubThumbnail />,
        url: GITHUB_ADRESS,
    },
    {
        id: 2,
        testId: "linkedin-icon-test",
        icon: <LinkedinThumbnail />,
        url: LINKEDIN_ADRESS,
    },
    {
        id: 3,
        testId: "upwork-icon-test",
        icon: <UpworkThumbnail />,
        url: UPWORK_ADRESS,
    },
];