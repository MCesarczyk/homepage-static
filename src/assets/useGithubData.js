import { useEffect, useState } from "react";
import { fetchAPIData } from "./fetchAPIData";

export const useGithubData = () => {
  const [repos, setRepos] = useState({
    status: "loading",
    repos: null
  });

  const apiURL = 'https://api.github.com/users/MCesarczyk/repos?sort="updated"';

  useEffect(() => {
    const getRepos = async () => {
      try {
        const repos = await fetchAPIData(apiURL);

        if (!repos) {
          throw new Error("No repo data");
        }

        setRepos({
          status: "success",
          repos: repos
        });

      } catch (error) {
        setRepos({ status: "error" });
        console.log(error.message);
      }
    };

    setTimeout(getRepos, 2_000);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return repos;
};