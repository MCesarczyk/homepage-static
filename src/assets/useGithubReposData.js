import { useEffect, useState } from "react";
import { fetchAPIData } from "./fetchAPIData";

export const useGithubReposData = () => {
  const [repos, setRepos] = useState({
    status: "loading",
    repos: []
  });

  const DEMO_DELAY = 2_500;

  const apiURL = 'https://api.github.com/users/MCesarczyk/repos?sort="updated"';

  useEffect(() => {
    const getRepos = async () => {
      try {
        const repos = await fetchAPIData(apiURL);

        if (!repos) {
          throw new Error("No repo data");
        }

        setRepos({
          status: "success",
          repos: repos
        });

      } catch (error) {
        setRepos({ status: "error" });
        console.error(error.message);
      }
    };

    setTimeout(getRepos, DEMO_DELAY);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return repos;
};