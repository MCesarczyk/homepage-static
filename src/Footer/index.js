import { MAIL_ADRESS } from "../assets/adresses";
import "./style.css";

const Footer = () => (
    <div className="footer__wrapper">
        <h2 className="footer__header">
            <p className="footer__caption">
                CONTACT:
            </p>
            <a className="footer__link" id="contact" href={`mailto:${MAIL_ADRESS}`}>
                {MAIL_ADRESS}
            </a>
        </h2>
        <p className="footer__paragraph">
            I'm open to new challenging tasks, so when you need website or Single Page Application to launch, feel free to talk to me.
        </p>
    </div>
);

export default Footer;