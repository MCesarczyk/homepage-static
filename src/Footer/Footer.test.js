import { render, screen } from '@testing-library/react';
import { thumbnails } from '../assets/thumbnails';
import Footer from '.';

test('renders footer email', () => {
  render(<Footer />);
  const linkElement = screen.getByText(/michal.cesarczyk@gmail.com/i);
  expect(linkElement).toBeInTheDocument();
});

test('footer email renders properly', () => {
  render(<Footer />);
  const linkElement = screen.getByText(/michal.cesarczyk@gmail.com/i);
  expect(linkElement).toMatchSnapshot();
});

test('thumbnails must have a valid href', () => {
  render(<Footer />);

  thumbnails.forEach(thumbnail => {
    const icon = screen.getByTestId(thumbnail.testId);
    expect(icon).toHaveAttribute('href', thumbnail.url);
  });
});

test('thumbnails must render properly', () => {
  render(<Footer />);

  thumbnails.forEach(thumbnail => {
    const thumbnailItem = screen.getByTestId(thumbnail.testId).parentNode;
    expect(thumbnailItem).toMatchSnapshot();
  });
});