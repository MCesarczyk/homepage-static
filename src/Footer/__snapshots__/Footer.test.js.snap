// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer email renders properly 1`] = `
<a
  class="footer__link"
  href="mailto:michal.cesarczyk@gmail.com"
  id="contact"
>
  michal.cesarczyk@gmail.com
</a>
`;

exports[`thumbnails must render properly 1`] = `
<div
  class="footer__thumbnail"
>
  <a
    data-testid="github-icon-test"
    href="https://github.com/MCesarczyk"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="footer__icon"
    >
      <svg>
        githubThumbnail.svg
      </svg>
    </div>
  </a>
</div>
`;

exports[`thumbnails must render properly 2`] = `
<div
  class="footer__thumbnail"
>
  <a
    data-testid="linkedin-icon-test"
    href="https://www.linkedin.com/in/micha%C5%82-cesarczyk-0820589a"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="footer__icon"
    >
      <svg>
        linkedinThumbnail.svg
      </svg>
    </div>
  </a>
</div>
`;

exports[`thumbnails must render properly 3`] = `
<div
  class="footer__thumbnail"
>
  <a
    data-testid="upwork-icon-test"
    href="https://www.upwork.com/freelancers/~01ec95abd3619f24c6"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="footer__icon"
    >
      <svg>
        upworkThumbnail.svg
      </svg>
    </div>
  </a>
</div>
`;
